docs/02_database/05_database-interface/01_entitydb.mdx

```diff
@@ -662,62 +747,417 @@ final var subscription = db.subscribe(Trade.class)
+Never on the individual row level.
+
+The goal of this batching is to minimise work, and to publish updates that are accurate, rather than to publish every change.
+
+In this regard it is better to think in terms of a batching interval, rather than a batching period, which looks something like this:
+
+```
+...
+*
+|---- gathering phase ----| (batching period)
+*
+*
+*
+*
+*
+|---- flushing phase -----| (tick happens here)
+* aggregate updates
+* publish updates
+|---- gathering phase ----| (next batching period)
+*
+*
+...
+```
+
+### Batching is non-deterministic
+
+During our gathering phase, we listen for updates; and during the flushing phase, we aggregate and publish.
+
+Let's look at an example:
+
+Imagine we insert a trade with quantity 100. Within milliseconds, we update this trade and set the quantity to 200.
+
+Depending on when the tick happens, you could either get two updates:
+
+```
+|---- gathering phase ----|
+*
+*
+*
+*
+* insert (qty 100)
+|---- flushing phase -----|
+* aggregate updates
+* publish updates insert (qty 100)
+|---- gathering phase ----|
+* modify (qty 200)
+*
+*
+*
+*
+|---- flushing phase -----|
+* aggregate updates
+* publish updates modify (qty 200)
+|---- gathering phase ----| (batching period)
+*
+*
+```
+
+1. Insert trade, qty = 100
+2. Modify trade, qty = 200
+
+Or you could receive a single update:
+
+```
+|---- gathering phase ----|
+*
+*
+*
+* insert (qty 100)
+* modify (qty 200)
+|---- flushing phase -----|
+* aggregate updates
+* publish updates insert (qty 200)
+```
+
+1. Insert trade, qty = 200
+
+Depending on timing, either case is possible, and the smaller the gap between the insert and modify, the more likely the second scenario becomes.
+However, even if these operations happen immediately after each other, we could still receive two updates.
+
+When we get two updates of the same record within the batching interval we will always aggregate the updates.
+This could even mean that updates will be disregarded all together, for example if we receive an insert and a delete within the batching interval.
+
+This will always be non-deterministic, as we don't provide any guarantees.
+
+#### Batching at different levels
+
+The other thing to note is that batching happens on both the data server and the update queue level.
+We batch on the queue level to minimise the amount of work the data server has to do.
+Because we pre-allocate LMDB when the data sever starts, having too many needless updates would eat into that allocation.
+We batch on the data server to minimise the load on the client and the connection.
+
+### Out-of-sequence updates on the queue
+
+Updates are published on the update queue once a record has been written to the database, or when a transaction completes you are using a transaction database.
+
+Within a multi-node set-up, updates are published locally and forwarded to other nodes in genesis-cluster. As a result, sometimes a node can receive updates out of sequence.
```

Comment:
maybe worth explaining that if you use an external mqtt broker or JMS broker the chances of out of sequence updates will be lower, as there is a single centralised place to publish and consume data. It also depends on where the message broker is located, as a broker in the same availability zone as a genesis node will have different latency properties. 

When using message brokers genesis-cluster is not involved in terms of message forwarding.

Reply:
I have added this text

Reply:
looks good to me

Reply:
I'm not sure we should add that statement before we can verify that. We don't really know what the likelihood of out of sync messages is in ZeroMQ MQTT, or ActiveMQ. Untill we do, can we really say that this must be the case?